src/ssl_load.c

```diff
@@ -2875,9 +2875,22 @@ WOLFSSL_ABI
 int wolfSSL_CTX_load_verify_locations(WOLFSSL_CTX* ctx, const char* file,
                                      const char* path)
 {
-    /* Load using default flags/options. */
+    /* We want to keep trying to load more CA certs even if one cert in the
+     * directory is bad and can't be used (e.g. if one is expired), and we
+     * want to return success if any were successfully loaded (mimicking
+     * OpenSSL SSL_CTX_load_verify_locations() semantics), so we use
+     * WOLFSSL_LOAD_FLAG_IGNORE_ERR.  OpenSSL (as of v3.3.2) actually
+     * returns success even if no certs are loaded (e.g. because the
+     * supplied "path" doesn't exist or access is prohibited), and only
+     * returns failure if the "file" is non-null and fails to load.
+     *
+     * Note that if a file is supplied and can't be successfully loaded, the
+     * overall call fails and the path is never even evaluated.  This is
+     * consistent with OpenSSL behavior.
+     */
+
     int ret = wolfSSL_CTX_load_verify_locations_ex(ctx, file, path,
-        WOLFSSL_LOAD_VERIFY_DEFAULT_FLAGS);
+        WOLFSSL_LOAD_VERIFY_DEFAULT_FLAGS | WOLFSSL_LOAD_FLAG_IGNORE_ERR);
```

Comment:
`wolfSSL_CTX_load_verify_locations` is commonly used outside of OpenSSL compat builds. Its included in the default build. This change is detrimental to users relying on wolfSSL's current behaviour. If this change is necessary for an OSP, then I think its better to change the call to `wolfSSL_CTX_load_verify_locations_ex` and explicitly select the desired flags.

Reply:
Ah OK, was concerned that might be the case.  And it's labeled `WOLFSSL_ABI` so we're stuck continuing with existing semantics.  What we can do is map the compat layer version to the new implementation, under a new name, and restore the native implementation with its original name to assure no change in non-compat semantics

I've implemented this now, and when updating `openssl/ssl.h` I discovered
```
#ifdef WOLFSSL_APACHE_HTTPD
    #define SSL_CTX_load_verify_locations(ctx,file,path) \
        wolfSSL_CTX_load_verify_locations_ex(ctx,file,path,\
                                                   WOLFSSL_LOAD_FLAG_IGNORE_ERR)
#else
    #define SSL_CTX_load_verify_locations     wolfSSL_CTX_load_verify_locations
#endif
```
so this apparently has come up before.  I replaced all of that with
```
    #define SSL_CTX_load_verify_locations     wolfSSL_CTX_load_verify_locations_compat
```